src/en/readcomic/src/eu/kanade/tachiyomi/extension/en/readcomic/Readcomic.kt

```diff
@@ -0,0 +1,201 @@
+package eu.kanade.tachiyomi.extension.en.readcomic
+
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.source.model.Filter
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.ParsedHttpSource
+import okhttp3.Request
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+class Readcomic : ParsedHttpSource() {
+
+    override val name = "ReadComic"
+
+    override val baseUrl = "https://readcomic.net"
+
+    override val lang = "en"
+
+    override val supportsLatest = true
+
+    override fun popularMangaSelector() = ".manga-box"
+
+    override fun latestUpdatesSelector() = ".home-list .hl-box .hlb-name"
+
+    override fun popularMangaRequest(page: Int): Request {
+        return GET("$baseUrl/popular-comic/$page")
+    }
+
+    override fun latestUpdatesRequest(page: Int): Request {
+        return GET("$baseUrl/comic-updates/$page")
+    }
+
+    override fun popularMangaFromElement(element: Element): SManga =
+        SManga.create().apply {
+            setUrlWithoutDomain(element.selectFirst("a")!!.attr("abs:href"))
+            title = element.selectFirst("h3")!!.text()
+            thumbnail_url = element.selectFirst(".image")!!.selectFirst("img")!!.attr("abs:src")
+        }
+
+    override fun latestUpdatesFromElement(element: Element): SManga =
+        SManga.create().apply {
+            setUrlWithoutDomain(element.attr("abs:href"))
+            title = element.text()
+        }
+
+    override fun popularMangaNextPageSelector() = ".general-nav a:last-child"
+
+    override fun latestUpdatesNextPageSelector() = popularMangaNextPageSelector()
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request { // publisher > writer > artist + sorting for both if else
+        var statusFilterText = ""
+        var included = ""
+        var excluded = ""
+
+        if (!filters.isEmpty()) {
+            for (filter in filters) {
+                when (filter) {
+                    is StatusFilter -> {
+                        if (filter.state == 1) {
+                            statusFilterText = "CMP"
+                        } else if (filter.state == 2) statusFilterText = "ONG"
+                    }
+                    is GenreList -> {
+                        included = filter.included.joinToString(",")
+                        excluded = filter.excluded.joinToString(",")
+                    }
+
+                    else -> {}
+                }
+            }
+        }
+
+        return GET("https://readcomic.net/advanced-search?key=${query.replace(" ","+")}&wg=$included&wog=$excluded&status=$statusFilterText&page=$page")
```

Comment:
Use `baseUrl` and `.toHttpUrl().newBuilder()` so you can add properly encoded query parameters